### PR TITLE
feat(typography): add `monospace` styling and `Code` component

### PR DESCRIPTION
**Pull request by jzempel**
- [ ] **BREAKING CHANGE?** 

## Description

* Add `monospace` prop to existing `SM`, `MD`, and `LG` typography components
* Add new `Code` component with support for `size` and `type` color styling

## Detail

Demo pre-published for review: https://garden.zendesk.com/react-components/typography/

## Checklist

- [x] :ok_hand: design updates are Garden Designer approved (add the
      designer as a reviewer)
- [x] :nail_care: view component styling is based on a Garden CSS
      component
- [x] :globe_with_meridians: Styleguidist demo is up-to-date (`yarn start`)
- [x] :arrow_left: renders as expected with reversed (RTL) direction
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [x] :guardsman: includes new unit tests
- [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: packages/typography/src/views/Code.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'styled-components';
+import {
+  zdColorGreen200,
+  zdColorGreen700,
+  zdColorGrey200,
+  zdColorGrey700,
+  zdColorRed300,
```
[the remark this answers is not in the document]
Comment (vedranio) [reply packages/typography/src/views/Code.js]: Sorry, ignore my spelling in all the color suggestions. It should be `zdColorRed200`

File: packages/typography/src/views/Code.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+const codeAttributes = {
+  'data-garden-id': COMPONENT_ID,
+  'data-garden-version': PACKAGE_VERSION
+};
+
+const codeCSS = css`
+  border-radius: 2px;
+  background-color: ${props => backgroundColor(props.type)};
+  padding: 1.5px 2px;
```
[the remark this answers is not in the document]
Comment (jzempel) [reply packages/typography/src/views/Code.js]: @vedranio thanks for the precision review here. I've corrected to use `zdColorRed200`.

re: the vertical padding... In the designs, I'm seeing container heights that are less than the content line heights (for every size variation). For the `inline` text CSS treatment, that means padding the `font-size`. So here's how I arrived at the values:
* SM: 11px font-height + 3px padding = 14px "container" height (vs 16px line-height)
* MD: 13px font-height + 3px padding = 16px "container" height (vs 20px line-height)
* LG: 17px font-height + 3px padding = 20px "container" height (vs 24px line-height)

Screen Shot 2019-04-12 at 8 01 09 AM

re: horizontal padding... as you can see from the screenshot, the text naturally has leading/trailing space within its boundary. This is where I came up with the 2px – one for the intended container pixel + another for the naturally occurring space in the design tool.

I'm not exactly sure what padding values will work best. Are you thinking `padding: 1px 1px` or maybe `padding 1px 1.5px`? Note these values give us an odd-valued visual height for the code fragments.

Would you be willing to play with the demo + dev tools and let me know what CSS combinations render closest to your intended designs? Please note, we can apply different paddings for each size, if that is what's needed.

[the remark this answers is not in the document]
Comment (jzempel) [reply packages/typography/src/views/Code.js]: @vedranio fwiw I updated the demo with `yellow-800` foreground and padding of 1px (vertical) and 1.5px (horizontal) as an attempt to nudge this in the right direction. Once you've played around with the CSS in dev tools, LMK how you'd like to proceed.

File: packages/typography/src/views/Code.spec.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React from 'react';
+import { mount } from 'enzyme';
+import { mountWithTheme } from '@zendeskgarden/react-testing';
```
Comment (austin94) [inline packages/typography/src/views/Code.spec.js]: [nit] could this all be `mountWithTheme`?

File: packages/typography/src/views/Code.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (vedranio) [reply packages/typography/src/views/Code.js]: @jzempel once you explained the logic, it makes complete sense.
I just played with it in code and the difference is soooo minor that I'd be happy to go with your original vertical padding which also avoids the odd-valued visual height.

For the horizontal padding, 1.5px works

So let's go with `padding: 1.5px 1.5px`

[the remark this answers is not in the document]
Comment (jzempel) [reply packages/typography/src/views/Code.js]: Thanks @vedranio – updated to your specs.

**Pull request merged**